mgr/src/server/plugins/volume_heal.cr

```diff
@@ -0,0 +1,168 @@
+      storage_unit.heal_metrics.heal_possibly_healing_count = heal_storage_unit_data.heal_possibly_healing_count
+      storage_unit.heal_metrics.heal_pending_files = heal_storage_unit_data.heal_pending_files
+    end
+  end
+
+  volume
+end
+
+post "/api/v1/pools/:pool_name/volumes/:volume_name/heal/start" do |env|
+  pool_name = env.params.url["pool_name"]
+  volume_name = env.params.url["volume_name"]
+
+  next forbidden(env) unless Datastore.maintainer?(env.user_id, pool_name)
+
+  pool = Datastore.get_pool(pool_name)
+  if pool.nil?
+    halt(env, status_code: 400, response: ({"error": "The Pool(#{pool_name}) doesn't exists"}.to_json))
+  end
+
+  volume = Datastore.get_volume(pool_name, volume_name)
+  if volume.nil?
+    halt(env, status_code: 400, response: ({"error": "The Volume(#{volume_name}) doesn't exists"}.to_json))
+  end
+
+  if !volume.replicate_family?
+    halt(env, status_code: 400, response: ({"error": "Cannot heal Volume(#{volume_name}). It is non replicated"}.to_json))
+  end
+
+  client_volfile = "/var/lib/kadalu/volfiles/client-dev-#{volume_name}.vol"
+
+  if !File.exists?(client_volfile)
+    volfile_name = "client"
+    tmpl = volfile_get("client")
+    content = Volfile.volume_level(volfile_name, tmpl, volume)
+    File.write(client_volfile, content)
+  end
+
+  rc, output, err = execute("/usr/local/libexec/glusterfs/glfsheal", ["vol1", "--xml", "volfile-path", client_volfile])
+
+  if rc < 0
+    halt(env, status_code: 400, response: ({"error": err}.to_json))
+  end
+
+  document = XML.parse(output)
+
+  brick_data = parse_heal_data(document)
+  volume = set_heal_data(volume, brick_data)
+
+  env.response.status_code = 201
+  volume.to_json
+end
+
+get "/api/v1/pools/:pool_name/volumes/:volume_name/heal" do |env|
+  pool_name = env.params.url["pool_name"]
+  volume_name = env.params.url["volume_name"]
+
+  next forbidden(env) unless Datastore.maintainer?(env.user_id, pool_name)
+
+  pool = Datastore.get_pool(pool_name)
+  if pool.nil?
+    halt(env, status_code: 400, response: ({"error": "The Pool(#{pool_name}) doesn't exists"}.to_json))
+  end
+
+  volume = Datastore.get_volume(pool_name, volume_name)
+  if volume.nil?
+    halt(env, status_code: 400, response: ({"error": "The Volume(#{volume_name}) doesn't exists"}.to_json))
+  end
+
+  if !volume.replicate_family?
+    halt(env, status_code: 400, response: ({"error": "Cannot heal Volume(#{volume_name}). It is non replicated"}.to_json))
+  end
+
+  client_volfile = "/var/lib/kadalu/volfiles/client-dev-#{volume_name}.vol"
+
+  if !File.exists?(client_volfile)
+    volfile_name = "client"
+    tmpl = volfile_get("client")
+    content = Volfile.volume_level(volfile_name, tmpl, volume)
+    File.write(client_volfile, content)
+  end
+
+  rc, output, err = execute("/usr/local/libexec/glusterfs/glfsheal", ["vol1", "info-summary", "--xml", "volfile-path", client_volfile])
```

Comment:
same here, make `glfsheal_bin` as a function and add the code as shown in previous comment.